### ISSUE TITLE
Allow adding custom types to _native_to_marshmallow

### ISSUE THREAD

**Issue by rakanalh** [lovasoa/marshmallow_dataclass#16]
Follow up to #15, this is to suggest exposting `_native_to_marshmallow` publicly or, better yet, allow registring custom types to the list for more flexibility on the types that could be user-defined and which fields these types should translate to.

**Comment by lovasoa** [comment]
The supported way to use custom fields for your type is to add a `marshmallow_field` key to your [field metadata](https://docs.python.org/3/library/dataclasses.html#dataclasses.field), and make it contain your marshmallow field.

**Comment by rakanalh** [comment]
I did see it in the documentation. The reason i wanted to not do this is because of what i wrote in a comment in a different issue:
https://github.com/lovasoa/marshmallow_dataclass/issues/15#issuecomment-490617905

**Comment by evanfwelch** [comment]
I would also find this valuable... a mapping you can can pass to `class_schema` that gets merged with `_native_to_marshmallow`.

```
    if typ in _marshmallow_mapping:
        field_fn = _marshmallow_mapping[typ]
        if inpsect.isfunction(field_fn):
            return field_fn(typ, **metadata)
        else: #assume it's just a marshmallow field
             return field_fn(**metadata)
```

**Comment by alextremblay** [comment]
There was an idea introduced in another issue (#23) That I think could work very well here:

```python
from typing import List
from ipaddress import IPv4Address

from marshmallow.fields import Field
from marshmallow_dataclass import dataclass, NewType

class IPField(Field):
    ... # serializes an IPv4Address instance into a str, and deserializes back into an instance

#IP = NewType('IP', str, validate=my_validation_function) 
#This was the suggestion in the other issue

IP = NewType('IP', IPv4Address, marshmallow_field=IPField)

@dataclass
class Company:
    name: str
    ipv4s: List[IP]
```

The idea is to have a NewType implementation that works the same as typing.NewType, but also stores all the field metadata that we would normally store on a dataclass field, like validation functions and marshmallow_field selections.

Thoughts?

**Comment by lovasoa** [comment]
I just published version `6.1.0rc1` with support for `NewType`. You can check out [the README](https://github.com/lovasoa/marshmallow_dataclass#custom-newtype-declarations), the [documentation](https://github.com/lovasoa/marshmallow_dataclass#custom-newtype-declarations), and try it. If you encounter any issues, or find that something could be improved in the documentation, open an issue. If no one complains, then I'll publish it as `6.1.0` and close this issue.

**Comment by lovasoa** [comment]
Version 6.1.0 was just released with support for NewType, so I am closing this.
https://pypi.org/project/marshmallow-dataclass/6.1.0/
